Lib/pdb.py

```diff
@@ -1521,6 +1521,24 @@ def lookupmodule(self, filename):
                 return fullname
         return None
 
+    def _runmodule(self, module_name):
+        self._wait_for_mainpyfile = True
+        self._user_requested_quit = False
+        import runpy
+        mod_name, mod_spec, code = runpy._get_module_details(module_name)
+        self.mainpyfile = self.canonic(code.co_filename)
+        import __main__
+        __main__.__dict__.clear()
+        __main__.__dict__.update({
+            "__name__": "__main__",
+            "__file__": self.mainpyfile,
+#            "__package__": module_name,  # Not needed, will rely on __spec__.parent
```

Comment:
Even though it's not always needed these days, `runpy` still promises to set `__package__`, so this should set it too.